api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -389,3 +397,463 @@ func (m *AzureManagedMachinePool) validateKubeletConfig() error {
 	}
 	return nil
 }
+
+// validateLinuxOSConfig enforces AKS API configuration for Linux OS custom configuration
+// See: https://learn.microsoft.com/en-us/azure/aks/custom-node-configuration#linux-os-custom-configuration for detailed information.
+func (m *AzureManagedMachinePool) validateLinuxOSConfig() error {
+	if m.Spec.LinuxOSConfig == nil {
+		return nil
+	}
+
+	if m.Spec.LinuxOSConfig.SwapFileSizeMB != nil {
+		minSwapFileSizeMB := int32(1)
+		if *m.Spec.LinuxOSConfig.SwapFileSizeMB < minSwapFileSizeMB {
+			return field.Invalid(
+				field.NewPath("Spec", "LinuxOSConfig", "SwapFileSizeMB"),
+				m.Spec.LinuxOSConfig.SwapFileSizeMB,
+				fmt.Sprintf("SwapFileSizeMB must be at least %d", minSwapFileSizeMB))
+		}
+	}
+
+	if m.Spec.LinuxOSConfig.TransparentHugePageDefrag != nil {
+		switch *m.Spec.LinuxOSConfig.TransparentHugePageDefrag {
+		case TransparentHugePageAlways:
+		case TransparentHugePageDefer:
+		case TransparentHugePageDeferMadvise:
+		case TransparentHugePageMadvise:
+		case TransparentHugePageNever:
+		default:
+			return field.Invalid(
+				field.NewPath("Spec", "LinuxOSConfig", "TransparentHugePageDefrag"),
+				m.Spec.LinuxOSConfig.TransparentHugePageDefrag,
+				fmt.Sprintf("%s, is not a supported configuration", *m.Spec.LinuxOSConfig.TransparentHugePageDefrag))
+		}
+	}
+
+	if m.Spec.LinuxOSConfig.TransparentHugePageEnabled != nil {
+		switch *m.Spec.LinuxOSConfig.TransparentHugePageEnabled {
+		case TransparentHugePageAlways:
+		case TransparentHugePageMadvise:
+		case TransparentHugePageNever:
+		default:
+			return field.Invalid(
+				field.NewPath("Spec", "LinuxOSConfig", "TransparentHugePageEnabled"),
+				m.Spec.LinuxOSConfig.TransparentHugePageEnabled,
+				fmt.Sprintf("%s, is not a supported configuration", *m.Spec.LinuxOSConfig.TransparentHugePageEnabled))
+		}
+	}
+
+	if m.Spec.LinuxOSConfig.Sysctls != nil {
+		if fileSystemSettingsErrors := m.validateLinuxOSConfigFS(); fileSystemSettingsErrors != nil {
+			return fileSystemSettingsErrors
+		}
+
+		if networkCoreSettingsErrors := m.validateLinuxOSConfigNetworkCore(); networkCoreSettingsErrors != nil {
+			return networkCoreSettingsErrors
+		}
+
+		if networkIPv4SettingsErrors := m.validateLinuxOSConfigNetworkIPv4(); networkIPv4SettingsErrors != nil {
+			return networkIPv4SettingsErrors
+		}
+
+		if networkIPv4NeighSettingsErrors := m.validateLinuxOSConfigNetworkIPv4Neigh(); networkIPv4NeighSettingsErrors != nil {
+			return networkIPv4NeighSettingsErrors
+		}
+
+		if networkNetfilterSettingsErrors := m.validateLinuxOSConfigNetworkNetfilter(); networkNetfilterSettingsErrors != nil {
+			return networkNetfilterSettingsErrors
+		}
+
+		if kernelSettingsErrors := m.validateLinuxOSConfigKernel(); kernelSettingsErrors != nil {
+			return kernelSettingsErrors
+		}
+
+		if virtualMemorySettingsErrors := m.validateLinuxOSConfigVM(); virtualMemorySettingsErrors != nil {
+			return virtualMemorySettingsErrors
+		}
+	}
```

Comment:
I split out the `LinuxOSConfig.Sysctls` checks into multiple logical groupings to satisfy `golang-ci-lint` complaining about `cyclomatic complexity` being too high`106`.